### ISSUE TITLE
How to use for graph level unsupervised clustering, not the node?

### ISSUE THREAD

**Issue by PeterDeSOM** [LeonResearch/WARGA#1]
I have millions of graph data that cannot be labeled by human efforts. I need it to be classified by unsupervised technique, not semi and not the node level. Is it possible?

**Comment by LeonResearch** [comment]
Yes, it is possible. I would recommend you to check out papers on **Graph Self-Supervised Learning**, and in particular, the graph-level ones: 

1. GraphCL: https://github.com/Shen-Lab/GraphCL
2. InfoGraph: https://github.com/sunfanyunn/InfoGraph

Both papers use contrastive learning methods but from different perspectives. In short, they first pre-train a GNN backbone using a self-supervised loss function without any label information, then use the embeddings to perform downstream tasks like classification and clustering.

But in this paper, WARGA only considers node-level embedding learning and is not suitable for graph-level tasks.
